flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/TestingFlinkService.java

```diff
@@ -448,21 +435,46 @@ public List<String> getDisposedSavepoints() {
 
     @Override
     public Optional<Savepoint> getLastCheckpoint(JobID jobId, Configuration conf) throws Exception {
+        jobs.stream()
+                .filter(js -> js.f1.getJobId().equals(jobId))
+                .findAny()
+                .ifPresent(
+                        t -> {
+                            if (!t.f1.getJobState().isGloballyTerminalState()) {
```

Comment:
It's just a simple way to ensure that we don't build operator logic relying on last checkpoints when the job is not in a terminal state (because it's not actually guaranteed to be the latest in that case).